protocols/identify/src/handler.rs

```diff
@@ -318,11 +343,20 @@ impl ConnectionHandler for Handler {
         {
             self.inbound_identify_push.take();
 
-            if let Ok(info) = res {
-                self.update_supported_protocols_for_remote(&info);
-                return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(Event::Identified(
-                    info,
-                )));
+            if let Ok(remote_info) = res {
+                self.update_supported_protocols_for_remote(&remote_info);
+                match self.update_remote_public_key(remote_info).try_into() {
```

Comment:
Why only update the public key? Shouldn't we always fill in everything from the last `Info` that we received?

1. I'd suggest that we have a field `remote_info: Option<Info>`.
2. We update this field every time we fetch a new identify info from the remote.
3. Upon receiving a push, we compute a new `Info` based on the existing `Info` and the received "patch".
4. If we don't have an old `Info`, log a warning.
5. After patching, update the local field to the newly computed `Info`. Note that we should also call `update_supported_protocols_for_remote` with the new info.
6. Emit an event with the new info.